core/data/src/main/java/com/google/samples/apps/nowinandroid/core/data/util/ConnectivityManagerNetworkMonitor.kt

```diff
@@ -35,18 +35,25 @@ import kotlinx.coroutines.flow.conflate
 class ConnectivityManagerNetworkMonitor @Inject constructor(
     @ApplicationContext private val context: Context
 ) : NetworkMonitor {
-    override val isOnline: Flow<Boolean> = callbackFlow<Boolean> {
+    override val isOnline: Flow<Boolean> = callbackFlow {
+        val connectivityManager = context.getSystemService<ConnectivityManager>()
+
         val callback = object : NetworkCallback() {
```

Comment:
I think it's worth adding a comment why we need to call `connectivityManger.isCurrentlyConnected` for each method in the callback as it's not immediately obvious (as we found from trying fix #425).